android/src/main/java/com/thebluealliance/androidclient/gcm/GCMMessageHandler.java

```diff
@@ -80,6 +86,25 @@ private void getComponenet() {
         }
     }
 
+    /**
+     * Checks if the user subscribes to the given team. Implements FollowsChecker.
+     *
+     * @param teamNumber the team number without the "frc" prefix.
+     * @param notificationType one of NotificationTypes.UPCOMING_MATCH, MATCH_SCORE, ...
+     */
+    public boolean followsTeam(Context context, String teamNumber, String notificationType) {
+        String currentUser = AccountHelper.getSelectedAccount(context);
+        String teamKey = TeamHelper.baseTeamKey("frc" + teamNumber);
```

Comment:
There's one case you missed here. You'll also need to account for Team@Event subscriptions. The model key is of the format `2015onwa_frc1114`